bin/runtime-common/src/messages_batch_extension.rs

```diff
@@ -69,70 +200,98 @@ where
 		_call: &Self::Call,
 		_post_info: &DispatchInfoOf<Self::Call>,
 		_len: usize,
-	) -> Result<(), TransactionValidityError> {
+	) -> Result<Self::Pre, TransactionValidityError> {
 		// TODO: for every call from the batch - call the `BridgesExtension` to ensure that every
 		// call transaction brings something new and reject obsolete transactions
-		Ok(())
+		unimplemented!("TODO") // TODO: return actual call type
 	}
 
 	fn post_dispatch(
-		_pre: Option<Self::Pre>,
-		_info: &DispatchInfoOf<Self::Call>,
-		_post_info: &PostDispatchInfoOf<Self::Call>,
-		_len: usize,
-		_result: &DispatchResult,
+		pre: Option<Self::Pre>,
+		info: &DispatchInfoOf<Self::Call>,
+		post_info: &PostDispatchInfoOf<Self::Call>,
+		len: usize,
+		result: &DispatchResult,
 	) -> Result<(), TransactionValidityError> {
-		// TODO:
-		// if batch matches!(relay-chain-finality-call, parachain-finality-call,
-		// deliver-messages-call) then:
-		//
-		// 0) ensure that the `result` is `Ok(_)`
-		//
-		// 1) ensure that all calls are dependent (relaychain -> parachain of this relaychain and
```

Comment:
I don't see an explicit check that all inner batched calls are dependent (`relay_best`, `new_parahead` using `relay_best`, `msg_proof` using `new_parahead`).
Do you plan on checking this using yet unimplemented `fn relay_chain_state()` and `fn parachain_state()`?

Or do you want to drop this dependency check and allow batching unrelated/non-dependent relay/para/msgs? I mean, as long as they are valid, we don't have to force them dependent (but might end up more complex than actually expecting and checking they're dependent).

Reply:
Yep - it isn't implemented yet